main/modes/utilities/ScreenSaver/screensaver.c

```diff
@@ -0,0 +1,356 @@
+// Includes
+//==============================================================================
+
+#include "screensaver.h"
+#include "esp_random.h"
+#include "modeIncludeList.h"
+
+//==============================================================================
+// Defines
+//==============================================================================
+
+#define SUBPIXEL_COUNT 256
+#define MAX_VELOCITY   (8 * SUBPIXEL_COUNT)
+
+//==============================================================================
+// Consts
+//==============================================================================
+
+const char modeName[] = "Screensaver";
+
+/// @brief Contains a sprite for each major mode
+static const cnfsFileIdx_t items[] = {
+    MAG_FEST_BOUNCER_WSG,
+    BARREL_1_WSG,
+    MMX_TROPHY_WSG,
+};
+
+//==============================================================================
+// Enums
+//==============================================================================
+
+typedef enum
+{
+    LED_NORTH,
+    LED_WEST,
+    LED_SOUTH,
+    LED_EAST,
+} LEDDirections_t;
+
+//==============================================================================
+// Function declarations
+//==============================================================================
+
+static void screenEnterMode(void);
+static void screenExitMode(void);
+static void screenMainLoop(int64_t elapsedUs);
+static void updateObjects(void);
+static void lightLEDs(LEDDirections_t dir);
+static void fadeLEDs(void);
+static void colorExplosion(void);
+static void drawScreenSaver(void);
+
+//==============================================================================
+// Structs
+//==============================================================================
+
+typedef struct
+{
+    vec_t pos;         // Position and bounding box
+    vec_t velocity;    // current velocity
+    vec_t accumulator; // Velocity added
+    wsg_t image;       // Loaded WSG
+    bool isActive;     // Should draw
+} bouncingObject_t;
+
+typedef struct
+{
+    bouncingObject_t objs[ARRAY_SIZE(items)];
+    led_t leds[CONFIG_NUM_LEDS];
+    bool explosion;
+    int64_t explosionTimer;
+} screenSaverData_t;
+
+//==============================================================================
+// Variables
+//==============================================================================
+
+swadgeMode_t screenSaverMode = {
+    .modeName          = modeName,
+    .wifiMode          = NO_WIFI,
+    .overrideUsb       = false,
+    .usesAccelerometer = false,
+    .overrideSelectBtn = false,
+    .fnEnterMode       = screenEnterMode,
+    .fnExitMode        = screenExitMode,
+    .fnMainLoop        = screenMainLoop,
+};
+
+screenSaverData_t* ssd;
+
+//==============================================================================
+// Functions
+//==============================================================================
+
+static void screenEnterMode(void)
```

Comment:
If there's no sound, might as well call `setDacShutdown(true)` to get rid of DAC hum.